Modules/Messages/src/Messages.Application/Messages/Queries/GetMessage/Handler.cs

```diff
@@ -10,27 +11,24 @@ public class Handler : IRequestHandler<GetMessageQuery, MessageDTO>
 {
     private readonly IMessagesRepository _messagesRepository;
     private readonly IMapper _mapper;
-    private readonly MessageService _messageService;
     private readonly IUserContext _userContext;
 
-    public Handler(IUserContext userContext, IMapper mapper, MessageService messageService, IMessagesRepository messagesRepository)
+    public Handler(IUserContext userContext, IMapper mapper, IMessagesRepository messagesRepository)
     {
         _userContext = userContext;
         _mapper = mapper;
-        _messageService = messageService;
         _messagesRepository = messagesRepository;
     }
 
     public async Task<MessageDTO> Handle(GetMessageQuery request, CancellationToken cancellationToken)
     {
-        var message = await _messagesRepository.Find(request.Id, _userContext.GetAddress(), cancellationToken, true);
-        
-        await _messageService.MarkMessageAsReceived(message);
+        var message = await _messagesRepository.Find(request.Id, _userContext.GetAddress(), cancellationToken, track: true);
 
-        var response = _mapper.Map<MessageDTO>(message);
+        message.Recipients.FirstWithIdOrDefault(_userContext.GetAddress()).FetchedMessage(_userContext.GetDeviceId());
+
+        await _messagesRepository.Update(message);
 
-        if (!request.NoBody)
-            await _messageService.FillBody(response);
```

Comment:
You just deleted this part of the code. So now, if someone passes `NoBody = true`, they still receive the body. 

The `MessagesRepository.Find` method needs a parameter for defining whether the body should be returned or not.